### PR TITLE
1781 split eddi project

### PR DESCRIPTION
**Pull request by GioVAX**
This should include the bulk of file modifications.

The next stage will be removing references to namespace EDDI from namespace EddiCore.

Step 1 for issue #1781.

### CODE REVIEW

File: EDDI/EDDI.cs [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
+namespace EddiCore
 {
-    /// <summary>
-    /// Eddi is the controller for all EDDI operations.  Its job is to retain the state of the objects such as the commander, the current system, etc.
-    /// and keep them up-to-date with changes that occur.  It also passes on messages to responders to handle as required.
-    /// </summary>
-    public class EDDI
+	/// <summary>
+	/// Eddi is the controller for all EDDI operations.  Its job is to retain the state of the objects such as the commander, the current system, etc.
+	/// and keep them up-to-date with changes that occur.  It also passes on messages to responders to handle as required.
```
Comment (Tkael) [inline EDDI/EDDI.cs]: It acts as the switchboard for passing events through all parts of the application including both responders and monitors. :-)

File: EddiCore/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+[assembly: AssemblyProduct("EddiCore")]
+[assembly: AssemblyCopyright("Copyright ©  2020")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
```
Comment (Tkael) [inline EddiCore/Properties/AssemblyInfo.cs:21]: Please add (to help enable localization):
```
//In order to begin building localizable applications, set 
//<UICulture>CultureYouAreCodingWith</UICulture> in your .csproj file
//inside a <PropertyGroup>.  For example, if you are using US english
//in your source files, set the <UICulture> to en-US.  Then uncomment
//the NeutralResourceLanguage attribute below.  Update the "en-US" in
//the line below to match the UICulture setting in the project file.

[assembly: NeutralResourcesLanguage("en", UltimateResourceFallbackLocation.MainAssembly)]
```

**Comment by Tkael** [comment]
Need to add EddiCore.csproj to EDDI.sln?

File: EDDI/App.xaml.cs [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
 using System.Diagnostics;
 using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Security.Permissions;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Windows;
+using EddiCore;
+using EddiSpeechService;
```
Comment (Tkael) [inline EDDI/App.xaml.cs]: Adds some "usings" which are currently not used. Will these be used later or should these be cleaned up?
- using System.Collections.Generic;
- using System.IO;
- using System.Reflection;
- using System.Security.Permissions;
- using EddiSpeechService;

File: EDDI/EDDI.cs (hunk 1 above)
Comment (GioVAX) [reply EDDI/EDDI.cs]: Amended.

File: EddiCore/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (GioVAX) [reply EddiCore/Properties/AssemblyInfo.cs:21]: Nice catch!
I left the project in the commit in preparation for moving the relevant classes.

Added

File: EDDI/App.xaml.cs (hunk 3 above)
Comment (GioVAX) [reply EDDI/App.xaml.cs]: Cleaned up and sorted.

**Review by Tkael — APPROVED** [review]

File: EDDI/App.xaml.cs [hunk 4]
```diff
@@ -8,10 +9,10 @@
 
 namespace Eddi
 {
-    /// <summary>
-    /// Interaction logic for App.xaml
-    /// </summary>
-    public partial class App : Application
+	/// <summary>
```
Comment (richardbuckle) [inline EDDI/App.xaml.cs:12]: Tabs again. I don't have a strong view on the subject, but we don't want a mix.

File: EDDI/EDDI.cs [hunk 5]
```diff
@@ -1,32 +1,31 @@
...
 
-namespace Eddi
+namespace EddiCore
 {
-    /// <summary>
-    /// Eddi is the controller for all EDDI operations.  Its job is to retain the state of the objects such as the commander, the current system, etc.
-    /// and keep them up-to-date with changes that occur.  It also passes on messages to responders to handle as required.
-    /// </summary>
-    public class EDDI
+	/// <summary>
```
Comment (richardbuckle) [inline EDDI/EDDI.cs:23]: Tabs

File: EddiCore/EddiCore.csproj [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{28DC25D7-4DFD-4A77-B4FF-B9BADB711ECB}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>EddiCore</RootNamespace>
+    <AssemblyName>EddiCore</AssemblyName>
+    <TargetFrameworkVersion>v4.7.1</TargetFrameworkVersion>
```
Comment (richardbuckle) [inline EddiCore/EddiCore.csproj:12]: We have moved to 4.8 while this was been worked on, let's update this one too.

File: EddiCore/EddiCore.csproj [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    <FileAlignment>512</FileAlignment>
+    <Deterministic>true</Deterministic>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
```
Comment (richardbuckle) [inline EddiCore/EddiCore.csproj:20]: I think we want `..\bin\Debug`

File: EddiCore/EddiCore.csproj [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
```
Comment (richardbuckle) [inline EddiCore/EddiCore.csproj:28]: I think we want `..\bin\Release`

File: EDDI/MainWindow.xaml.cs [hunk 9]
```diff
@@ -23,10 +24,10 @@
 
 namespace Eddi
 {
-    /// <summary>
-    /// Interaction logic for MainWindow.xaml
-    /// </summary>
-    public partial class MainWindow : Window
+	/// <summary>
```
Comment (richardbuckle) [inline EDDI/MainWindow.xaml.cs:27]: Tabs again

File: EDDI/MainWindow.xaml.cs [hunk 10]
```diff
@@ -868,7 +869,7 @@ private void betaProgrammeEnabled(object sender, RoutedEventArgs e)
             if (runBetaCheck)
             {
                 // Because we have changed to wanting beta upgrades we need to re-check upgrade information
-                EDDI.Instance.CheckUpgrade();
+	            EddiUpgrader.CheckUpgrade();
```
Comment (richardbuckle) [inline EDDI/MainWindow.xaml.cs:872]: Tabs

File: EDDI/MainWindow.xaml.cs [hunk 11]
```diff
@@ -988,7 +989,7 @@ private void companionAppClicked(object sender, RoutedEventArgs e)
             }
             else
             {
-                // Logout from the companion app and start again
+                // Logout from the companion EddiApplication and start again
```
Comment (richardbuckle) [inline EDDI/MainWindow.xaml.cs:992]: Aha, "the companion app" refers to the FDevs's defunct iOS app. Hence "the companion app API" or "cAPI" is how we often refer to the API in question.

I would say, "logout from the cAPI and start again"

File: EDDI/EddiUpgrader.cs [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+using System.IO;
+using System.Reflection;
+using System.Security.Permissions;
+using EddiCore;
+using EddiSpeechService;
+using Utilities;
+
+namespace Eddi
+{
+	public class EddiUpgrader
```
Comment (richardbuckle) [inline EDDI/EddiUpgrader.cs:13]: Tabs again

File: EDDI/EDDI.cs [hunk 13]
```diff
@@ -50,36 +49,33 @@ static EDDI()
             Directory.CreateDirectory(Constants.DATA_DIR);
         }
 
+        public static void Init(bool safeMode)
+        {
+	        if (instance == null)
```
Comment (richardbuckle) [inline EDDI/EDDI.cs:54]: tabs

File: EDDI/EDDI.cs [hunk 14]
```diff
@@ -122,20 +118,13 @@ public static EDDI Instance
         // The event queue
         public ConcurrentQueue<Event> eventQueue { get; private set; } = new ConcurrentQueue<Event>();
 
-        private EDDI()
+        private EDDI(bool safeMode)
         {
+	        running = !safeMode;
```
Comment (richardbuckle) [inline EDDI/EDDI.cs:123]: tabs

File: EDDI/EDDI.cs [hunk 15]
```diff
@@ -122,20 +118,13 @@ public static EDDI Instance
...
             try
             {
                 Logging.Info(Constants.EDDI_NAME + " " + Constants.EDDI_VERSION + " starting");
 
-                // Start by fetching information from the update server, and handling appropriately
```
Comment (richardbuckle) [inline EDDI/EDDI.cs:131]: Don't we need to invoke the updater at this point?

**Review by richardbuckle — CHANGES_REQUESTED** [review]
Great work! I hope it has given you a good tour of the code base.
This looks good for step 1.
I shall implement my requests myself, to save time.

**Comment by richardbuckle** [comment]
PS @GioVAX you should now have write access to this repo, so no more need to work in your own fork.

**Comment by richardbuckle** [comment]
Parking for rebase into main repo so I can add commits.

File: EDDI/EDDI.cs (hunk 15 above)
Comment (richardbuckle) [reply EDDI/EDDI.cs:131]: OK it is called from the App ctor

**Comment by richardbuckle** [comment]
Closing for the rebased #1815

**Pull request closed without merging**